appcues/src/main/java/com/appcues/Appcues.kt

```diff
@@ -271,9 +277,11 @@ public class Appcues internal constructor(internal val scope: AppcuesScope) {
      *  @param token A globally unique token that identifies this device to FCM.
      */
     public fun setPushToken(token: String?) {
-        storage.pushToken = token
+        if (token != storage.pushToken) {
```

Comment:
adding a safe-guard check, if token is the same as the existing we dont track DeviceUpdated